### PR TITLE
add sensor deregistration.

### PR DESCRIPTION
**Pull request by jhaynes**

### CODE REVIEW

File: recipes/sensor.rb [hunk 1]
```diff
@@ -58,6 +58,17 @@
   runit_service 'mss' do
     action :enable
   end
+
+  creds = data_bag_item('credentials', 'metaflows')
+
+  template '/etc/rc0.d/K01delete-metaflows-sensor' do
```
Comment (hatmatter) [inline recipes/sensor.rb]: K is executed on runlevel exit and S on runlevel enter right? Do you mean for this to run on entering the halt state?

File: templates/default/deregister-node.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    SALT=`echo -n "The seven words you can't say on television: ... $password" |md5sum`
+    ENCR=`/tmp/md5pass $password $SALT|head -1`
+    wget "https://nsm.metaflows.com/perl/getconf.pl?action=getsensors&p=$ENCR&m=$email" -O sensors.tmp -q 
+    res1=$?
+    res2=`head -1 sensors.tmp | awk '{print $1}'`;
+    rm -f sensors.tmp
+    if [ "$res2" != "none" ] && [ "$res1" == 0 ]; then
+        echo "user information validated"
+        loggedin=1
+		SALT=`echo -n "The seven words you can't say on television: ... $password" |md5sum`
```
Comment (hatmatter) [inline templates/default/deregister-node.erb]: is this indent level intentional?

File: templates/default/deregister-node.erb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		echo "Stopping and resetting sensor configuration"
+		/nsm/etc/mss.sh stop
+		echo "deleteing instance... this may take a few moments"
+        DELETESENSOR=`curl -s --form upload="@/tmp/mfd.reg" "https://nsm.metaflows.com/perl/getconf.pl?action=postaws"`
+		echo $DELETESENSOR
+		rm /tmp/mfd.reg
+    else
+        echo $res1
+        echo $res2
+        echo "Login failed, retrying..."
```
Comment (hatmatter) [inline templates/default/deregister-node.erb]: probably want to sleep for at least a little here

File: recipes/sensor.rb (hunk 1 above)
Comment (jhaynes) [inline recipes/sensor.rb]: Noah and I just learned that on halt K is run before S. And in the log upload stuff in heph, if it was S, it ran after things like networks were gone.

Comment (hatmatter) [inline recipes/sensor.rb]: Whoa

**Comment by hatmatter** [comment]
If you put a sleep statement in there then 🚢

**Pull request closed without merging**